learning-development/r.qmd

```diff
@@ -430,9 +430,20 @@ This can be done most easily by navigating in RStudio through `Tools > Global op
 
 ---
 
-The base R function of round() rounds 5's downwards. To round them upwards you can create a custom function like the one below:
+The base R function of `round()` can behave unusually due to the way numbers are stored within R itself. `round()` is supposed to use a 'banker's round', in which mid-point values (e.g. 0.5, 1.5, 2.5) are rounded to the closest even number. For example, using `round()`, 1.5 should round to 2, but 2.5 should also round to 2. This is different than the way in which SQL performs rounding, which is always to round numbers at the mid-way point up to the next number (e.g. 2.5 rounds to 3).
+
+However, when investigating numbers in R, you can see that they are not always stored in the way you might think. This is due to floating point numbers. Whilst humans use base 10 (numbers 0-9), computers store numbers in a binary (base 2) system as a combinations of 1s and 0s. When translating between base 2 and base 10, sometimes numbers can be not quite what we expect. For example, asking R to print the number 1.4 to 18 decimal places shows that it is actually stored as:
 
+``` {r round_floating_point, eval=FALSE}
+sprintf("%.18f", 1.4)
+[1] "1.399999999999999911"
 ```
+
+These slight differences in the floating point numbers behind the scenes can result in unusual rounding errors using round(). If you're interested, [there's more information available in these forum posts](https://stat.ethz.ch/pipermail/r-help/2007-August/138014.html). 
+ 
+To round numbers upwards in a way that is consistent with SQL and other programming languages, you can create a custom function like the one below:
```

Comment:
I think we should switch this up and recommend using the already created function from dfeR, rather than getting everyone to reproduce the custom function.

So switch this line to "To round numbers upwards..., we recommend using the roundFiveUp() function provided by the dfeR package." Could also potentially put roundFiveUp() as a link to the code in dfeR instead of recreating it here.